uxarray/grid/utils.py

```diff
@@ -169,3 +169,75 @@ def _newton_raphson_solver_for_gca_constLat(init_cart,
         _iter += 1
 
     return np.append(y_new, constZ)
+
+
+def _get_cartesian_face_edge_nodes(face_nodes_ind, face_edges_ind,
+                                   edge_nodes_grid, node_x_values,
+                                   node_y_values, node_z_values):
+    """Helper function that constructs the face-edge connectivity for a Cartesian grid.
+    Parameters
+    ----------
+    face_nodes_ind :
+        The ith entry of Grid.face_node_connectivity
+    face_edges_ind :
+        The ith entry of Grid.face_edge_connectivity
+    edge_nodes_grid :
+        The entire Grid.edge_node_connectivity
+    node_x_values :
+        The values of Grid.node_x
+    node_y_values :
+        The values of Grid.node_y
+    node_z_values :
+        The values of Grid.node_z
+    Returns
+    -------
+    face_edges : np.ndarray
+        Face edge connectivity for Cartesian grid
+    """
+
+    # Construct array to hold the face edge data
+
+    face_edges = np.zeros((len(face_edges_ind), 2), dtype=INT_DTYPE)
+
+    # Assign edge_nodes
+
+    for ind in range(0, len(face_edges_ind)):
+        edge_idx = face_edges_ind[ind]
+        if edge_idx != INT_FILL_VALUE:
+            edge_nodes = edge_nodes_grid.values[edge_idx]
+        else:
+            edge_nodes = [INT_FILL_VALUE, INT_FILL_VALUE]
+        face_edges[ind] = edge_nodes
+
+    # Sort edge nodes in counter-clockwise order
+
+    starting_two_nodes_index = [face_nodes_ind[0], face_nodes_ind[1]]
+    face_edges[0] = starting_two_nodes_index
+    for idx in range(1, len(face_edges)):
+        if face_edges[idx][0] == face_edges[idx - 1][1]:
+            continue
+        else:
+            # Swap the node index in this edge
+
+            temp = face_edges[idx][0]
+            face_edges[idx][0] = face_edges[idx][1]
+            face_edges[idx][1] = temp
```

Comment:
@hongyuchen1030 

Are the face edges not assumed to already be counter clockwise at this step?

Reply:
They are counter clockwise. This step is make sure all nodes are connected

```python
input (undirected):  
edge1: 0--3
edge2: 2--3
edge3:0--2

output(directed):
edge1: 0-->3
edge2: 3-->2
edge3: 2-->0
```
Of course for illustration purpose, I only use node index above to give the rough idea. In the codes, they're represented by the vertex cartesian coordinates

Reply:
Makes sense!